myapp.rb

```diff
@@ -6,186 +6,55 @@
 require 'securerandom'
 require 'erb'
 require 'byebug'
+require 'pg'
 
-#postgreSQLで出力テスト
-conn = PG.connect( dbname: 'shinatra_memoapp' )
-# @memos = conn.exec( "SELECT * FROM memodata" ) #これ自体がハッシュなので、@memos[0]とかで取り出せそう
+include ERB::Util
+conn = PG.connect(dbname: 'shinatra_memoapp')
 
-def load
-  JSON.parse(File.read('json/memo.json'), symbolize_names: true)
-end
-
-def store(memodata)
-  File.open('json/memo.json', 'w') do |file|
-   JSON.dump(memodata, file)
-  end
-end
-# 他に必要なもの
-  # メモリスト(トップページで一覧表示させるため)
-  # @memolist
-
-
-# top
 get '/memos' do
-  @memos = load()
-  @memo_id_array = @memos.keys
-  @array_size = @memo_id_array.size
-
+  @memos = conn.exec('SELECT * FROM memodata').to_a
   erb :index
 end
 
-  # top
-  get '/memos' do #ここわからんかも
-    memos = conn.exec( "SELECT * FROM memodata" )
-    # 展開してhash作り直せば良い？
-      # 一旦無視 @memolist = @memo.map{ |memo| "<a href=/memos/#{memo[0]}>#{html_escape(memo[1][:title])}</a>" }.join("<br>")
-    erb :index
-  end
+get '/memos/new' do
+  erb :new
+end
 
 post '/memos' do
-  @memos = load()
-  @memo = { title: params[:memo_title].to_s, text: params[:memo_text].to_s }
-  @memos[SecureRandom.uuid] = @memo
-  store(@memos)
+  id = SecureRandom.uuid.to_s
+  title = params[:memo_title]
+  text = params[:memo_text]
+  conn.exec_params('INSERT INTO memodata (id, title, sentence) VALUES ($1, $2, $3)', [id, title, text])
 
-  post '/memos' do
-    PostgreSQL.create
-    redirect '/memos'
-  end
+  redirect '/memos'
+end
 
-# show
 get '/memos/:id' do
   @memo_id = params[:id].to_sym
-  @memos = load()
-  @memo = @memos[@memo_id]
-  @title = @memo[:title]
-  @text = @memo[:text]
+  @memo = conn.exec_params('SELECT * FROM memodata WHERE id = $1', [@memo_id])
```

Comment:
このままだと`@memo`が配列になってerb側で`[0]`が必要になるので、

```suggestion
  @memo = conn.exec_params('SELECT * FROM memodata WHERE id = $1', [@memo_id])[0]
```

のようにしてこの時点で最初の1件を取ってくる方がいいですね。